### DISCUSSION TITLE
Working nice, but some questions...

### DISCUSSION THREAD

**Discussion by ruslanx3m** [kaliiiiiiiiii/Selenium-Driverless#4] [Q&A] [answered]
1. how to set a proxy with authentication ? 
2. how to set path to chromedriver in options ? 
3. how to set user-agent ?

can you give some example, 2 days from javascript, node js how I  came to python only for this lib ... 
it's passing cloudflare ... 
is a same or (puppeteer, playwright) .. in node js lib ?
thank you.

**Comment by ttlns** [comment, marked as answer]
1. use this package with [selenium-profiles](https://github.com/kaliiiiiiiiii/Selenium-Profiles) and [selenium-injector](https://github.com/kaliiiiiiiiii/Selenium-Injector):
proxies can be set in the profile or dynamically
```python
from selenium_profiles.webdriver import Chrome
from selenium_profiles.profiles import profiles

profile = profiles.Windows() # or .Android()
profile["proxy"] = {
  "proxy":"http://user1:pass1@example_host.com:41149"
  }

driver = Chrome(profile=profile, injector_options=True, driverless_options=True)

driver.profiles.proxy.set_single("http://user2:pass2@example_host.com:41149")
print(driver.profiles.proxy.proxy)

driver.quit()  # Execute on the End!
```

2. You can't as Selenium-Driverless connects to Chrome directly, without the use of `chromedriver`. Therefore, chromedriver isn't needed
3. use with selenium-profiles again. For possile profile-settings, have a look at [example-profile](https://github.com/kaliiiiiiiiii/Selenium-Profiles#profiles). Note that changing the UserAgent might make you getting detected

> can you give some example

a lot of the syntax is the same as with selenium itsself. Therefore you can have a look at its [documentation](https://selenium-python.readthedocs.io/getting-started.html) or [selenium-profiles-documentation](https://github.com/kaliiiiiiiiii/Selenium-Profiles) or [selenium-injector](https://github.com/kaliiiiiiiiii/Selenium-Injector) documentation

> it's passing cloudflare ...

yes, indeed, that's the point of this package.


> is a same or (puppeteer, playwright) .. in node js lib ?

Do you mean if there's a similar package for node-js (`javascript`) ? No, not as I know yet. It surely would be possible, but requires some development.

Note: feel free to ask further questions. Also, please make use of the Discussion section

**Reply by tankow79** [reply]
An error occurs:

```
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2022.2.3\plugins\python-ce\helpers\pydev\pydevd.py", line 1496, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2022.2.3\plugins\python-ce\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "D:\My_Sckript\Twitch-Channel-Points-Miner-v2-master\test\proxi_tets.py", line 9, in <module>
    driver = Chrome(profile=profile, injector_options=True, driverless_options=True)
  File "D:\My_Sckript\Twitch-Channel-Points-Miner-v2-master\venv\lib\site-packages\selenium_profiles\webdriver.py", line 155, in __init__
    super().__init__(**kwargs)
  File "D:\My_Sckript\Twitch-Channel-Points-Miner-v2-master\venv\lib\site-packages\selenium_driverless\sync\webdriver.py", line 10, in __init__
    super().__init__(options=options,max_ws_size=max_ws_size)
  File "D:\My_Sckript\Twitch-Channel-Points-Miner-v2-master\venv\lib\site-packages\selenium_driverless\webdriver.py", line 111, in __init__
    if not options.user_data_dir:
AttributeError: 'Options' object has no attribute 'user_data_dir'
```

Tested the code:
```
from selenium_profiles.webdriver import Chrome
from selenium_profiles.profiles import profiles

profile = profiles.Windows() # or .Android()
profile["proxy"] = {
  "proxy":"http://username:password@proxy_ip:proxy_port"
  }

driver = Chrome(profile=profile, injector_options=True, driverless_options=True)
# driver = Chrome(profile=profile, injector_options=True) # there was an attempt

driver.profiles.proxy.set_single("http://username:password@proxy_ip:proxy_port")
print(driver.profiles.proxy.proxy)

driver.quit()  # Execute on the End!
```

**Comment by ttlns** [comment]
Closing as it seems to be resolved
